parser-typechecker/src/Unison/TermPrinter.hs

```diff
@@ -195,11 +195,12 @@ pretty0
       elideFQN im $ PrettyPrintEnv.termName n (Referent.Con ref i CT.Data)
     Request' ref i -> styleHashQualified'' (fmt S.Request) $
       elideFQN im $ PrettyPrintEnv.termName n (Referent.Con ref i CT.Effect)
-    Handle' h body -> let (im', uses) = calcImports im body in
+    Handle' h body -> let (imBody, usesBody) = calcImports im body
+                          (imH, usesH) = calcImports im h in
       paren (p >= 2)
-        $ (fmt S.ControlKeyword "handle" `PP.hang` pretty0 n (ac 2 Normal im doc) h)
+        $ (fmt S.ControlKeyword "handle" `PP.hang` usesBody [pretty0 n (ac 2 Block imBody doc) body])
         <> PP.softbreak
-        <> (fmt S.ControlKeyword "in" `PP.hang` uses [pretty0 n (ac 2 Block im' doc) body])
+        <> (fmt S.ControlKeyword "with" `PP.hang` usesH [pretty0 n (ac 2 Block imH doc) h])
```

Comment:
I'm not confident about these changes. It seems to work, but I didn't look that closely at the `im`/`calcImports` stuff.